src/main/java/com/a2/backend/controller/ProjectController.java

```diff
@@ -177,4 +176,32 @@ public ProjectController(
         val comment = discussionService.createComment(discussionId, commentCreateDTO);
         return ResponseEntity.status(HttpStatus.OK).body(comment);
     }
+
+    @Secured({SecurityConstants.USER_ROLE})
+    @PutMapping("/highlight/{commentId}")
+    public ResponseEntity<?> highlightComment(@PathVariable UUID commentId) {
+        val comment = discussionService.changeCommentHighlight(commentId);
+        return ResponseEntity.status(HttpStatus.OK).body(comment);
+    }
+
+    @Secured({SecurityConstants.USER_ROLE})
+    @PutMapping("/hide/{commentId}")
+    public ResponseEntity<?> hideComment(@PathVariable UUID commentId) {
+        val comment = discussionService.changeCommentHidden(commentId);
+        return ResponseEntity.status(HttpStatus.OK).body(comment);
+    }
+
+    @Secured({SecurityConstants.USER_ROLE})
+    @GetMapping("/all-comments/{discussionId}")
+    public ResponseEntity<?> getCommentsInDiscussion(@PathVariable UUID discussionId) {
+        val comments = discussionService.getComments(discussionId);
+        return ResponseEntity.status(HttpStatus.OK).body(comments);
+    }
+
+    @Secured({SecurityConstants.USER_ROLE})
+    @GetMapping("/filtered-comments/{discussionId}")
+    public ResponseEntity<?> getFilteredCommentsInDiscussion(@PathVariable UUID discussionId) {
+        val comments = discussionService.getFilteredComments(discussionId);
+        return ResponseEntity.status(HttpStatus.OK).body(comments);
+    }
```

Comment:
I think this could have been just one `@GetMapping("comment/{discussionId}"` endpoint and then you could have applied the filter only when the logged user was not the owner.